game-core/src/main/java/games/strategy/engine/data/Attachable.java

```diff
@@ -2,6 +2,9 @@
 
 import java.util.Map;
 
+/**
+ * An entity that can host attachments.
```

Comment:
Implied in this interface contract is a string 'key', what is that key value supposed to represent?

(There are some other problems here, like why have a mutable API, and the stringly types, so I'd prefer to fix those, but in terms of the current scope the javadoc would do well to describe what convention is being used the String key values).